### PR TITLE
[yosys] enable libyoys, and several cleanups

### PR DESCRIPTION
**Pull request by sjkelly**
- enable libyosys
  - correct version to match upstream
  - whitespace fixes
  - fix Product array

### CODE REVIEW

**Comment by sjkelly** [comment]
BB doesn't seem to like `.so` on Apple. I suppose it is safe to just rename to `.dylib`, or is there something I am missing here?

**Comment by sjkelly** [comment]
Relevant target: https://github.com/YosysHQ/yosys/blob/83c0f82dc842fc859dfb4b19e766b23f965cfbb3/Makefile#L681-L686

File: Y/Yosys/build_tarballs.jl [hunk 1]
```diff
@@ -14,20 +14,25 @@ dependencies = [
     Dependency("boost_jll"; compat="=1.76.0"), # max gcc7
     Dependency("Readline_jll"),
     Dependency("Tcl_jll"),
-    Dependency("Zlib_jll"; compat="~1.2.11"),
-    Dependency("Libffi_jll"; compat="~3.2.2"),
+    Dependency("Zlib_jll"; compat="1.2.12"),
+    Dependency("Libffi_jll"; compat="3.2.2"),
```
Comment (giordano) [inline Y/Yosys/build_tarballs.jl:18]: Libffi _*must*_ stay with
```suggestion
    Dependency("Libffi_jll"; compat="~3.2.2"),
```
see #3024.  Compat on Zlib is fairly useless

Comment (sjkelly) [reply Y/Yosys/build_tarballs.jl]: Thanks! I've got that issue bookmarked now :)

**Pull request merged**